File: app-fw/app-base/src/main/java/org/yy/base/dao/dynamic2/WhereExpression.java
package org.yy.base.dao.dynamic2;

public class WhereExpression extends Expression {

	public WhereExpression() {
	}

	public String Eval(ExpressionContext ctx) {
		if (this.lists.size() == 0) {
			return "";
		}

		StringBuilder sb = new StringBuilder();
		String result = this.lists.get(0).Eval(ctx);
		if (result != null && result.trim().length() > 0) {
			sb.append(" WHERE ");
			sb.append(result);
		}

		return sb.toString();
	}
}

File: SqlParser/SqlParser/WhereExpression.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SqlParser
{
    public class WhereExpression : Expression
    {

        public WhereExpression()
        {
        }
        public override string Eval(IExpressionContext ctx)
        {
            if (this.lists.Count == 0 )
            {
                return "";
            }

            StringBuilder sb = new StringBuilder();
            string result = this.lists[0].Eval(ctx);
            if (result != null && result.Trim().Length > 0)
            {
                sb.Append(" WHERE ");
                sb.Append(result);
            }

            return sb.ToString();
        }
    }
}
